### DISCUSSION TITLE
Change class based on x-text value

### DISCUSSION THREAD

**Discussion by hajhamzah** [alpinejs/alpine#674] [4. General]
I render data from server using AlpineJS + Tailwindcss (both via CDN) as follow:

div class="w-full" x-data="loadGuru()" x-init="fetch('/api/guru/') .then(response => response.json()) .then(data => guru = data)"
...
template x-for="item in Guru" :key="item"
...
p class=px-2 inline-flex text-xs leading-5 font-semibold rounded-full bg-green-200 text-green-800" x-text="item.status" :class="{ 'bg-pink-500' : item.status == 'Libur' }
...
/template

Class of the item.status will be rendered as expected (background pink-500) if condition meet (item.status = Libur). But if i change the :class into, let say, :class="{bg-yellow-500' :item.status == 'Libur'}", it failed. If i refresh the page, it rendered as bg-green-200 text-green-800. Same result if i change the :class into, let say, bg-red-500, or bg-white. But if i change it into :class="{ 'bg-yellow-500 text-2xl' : item.status == 'Libur' }", item.status rendered as text-2xl but with background green (bg-green-200). Can someone help me on this issue? Thank you.

**Comment by SimoTod** [comment]
This is standard behaviour of css. :class just adds a css class to your element when a certain condition is met. You are telling the browser that your paragraph is both green and yellow so the browser has to make a decision. This decision follows the css order of precedence. Since bg-green-200 and bg-yellow-500 have the same importance, the one that is defined later in tailwind wins.
You need to use :class for both backgrounds:
`:class="{ 'bg-yellow-500' : item.status == 'Libur', 'bg-green-200' : item.status != 'Libur' }`

**Reply by hajhamzah** [reply]
Thank you @SimoTod. I got what i want by using :class for both background and text color:

```
<p class="px-2 inline-flex text-xs leading-5 font-semibold rounded-full" x-text="item.status" :class="{ 'bg-yellow-500 text-white' : item.status == 'Libur', 'bg-green-200 text-green-800' : item.status != 'Libur' }"></p>
```
